### PR TITLE
*: unify version and set max compatible via NextMajorVersion (#798)

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #798 to release-4.0
You can switch your code base to this Pull Request by using [git-extras](https://github.com/tj/git-extras):
```bash
# In br repo:
git pr https://github.com/pingcap/br/pull/852
```

After apply modifications, you can push your change to this PR via:
```bash
git push git@github.com:ti-srebot/br.git pr/852:release-4.0-9348911d9440
```

---

### What problem does this PR solve? 

Unify version and set max compatible via NextMajorVersion.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change

Related changes

 - Need to cherry-pick to the release branch

### Release Note

 - No release note.

### CODE REVIEW

File: pkg/lightning/backend/importer.go [hunk 1]
```diff
@@ -47,6 +50,15 @@ var (
...
 	requiredMaxTiKVVersion = *semver.New("5.0.0")
+=======
+	// Importer backend is compatible with TiDB [2.1.0, NextMajorVersion).
+	requiredMinTiDBVersion = *semver.New("2.1.0")
+	requiredMinPDVersion   = *semver.New("2.1.0")
+	requiredMinTiKVVersion = *semver.New("2.1.0")
+	requiredMaxTiDBVersion = version.NextMajorVersion()
+	requiredMaxPDVersion   = version.NextMajorVersion()
+	requiredMaxTiKVVersion = version.NextMajorVersion()
+>>>>>>> 9348911... *: unify version and set max compatible via NextMajorVersion (#798)
```
Comment (kennytm) [inline pkg/lightning/backend/importer.go:61]: merge conflict

**Review by overvenus — APPROVED** [review]

**Pull request merged**